src/malli/core.cljc

```diff
@@ -233,10 +234,30 @@
                     acc explainers)))))
           (-transformer [this transformer]
             (let [transformers (some->> entries
-                                        (mapcat (fn [[k _ s]] (if-let [t (-transformer s transformer)] [k t])))
+                                        (mapcat (fn [[k _ s]]
+                                                  (let [key-transformer (-transformer (map-key k) transformer)
+                                                        value-transformer (-transformer s transformer)
+                                                        noop-transformer (fn [x] x)]
+                                                    (cond
+                                                      (and key-transformer value-transformer)
+                                                      [k [key-transformer value-transformer]]
+
+                                                      (and (not key-transformer) value-transformer)
+                                                      [k [noop-transformer value-transformer]]
+
+                                                      (and key-transformer (not value-transformer))
+                                                      [k [key-transformer noop-transformer]]))))
```

Comment:
I think all keys should be transformed, not just the ones defined in the schema. e.g. the map might be open and the value can have extra keys, which should be transformed too.

Given that, I think the key-transformation could be an extra step, just like the `map-transformer`. With this, the performance for not using that would be identical as before. If both the `map-transformer` & `key-transformer` are present, there could be a max of 3 steps. I think it's ok, can't think of a way doing it faster.